docs/api/annotations.md

```diff
@@ -0,0 +1,233 @@
+`perPage`     | `string` | **Optional** Amount of annotations to return.
+`startCursor` | `string` | **Optional** A previously returned cursor to use as an 'offset'.
+
+```http
+GET /annotations HTTP/1.1
+Accept: application/json
+```
+
+### Response
+
+The response is a JSON list with annotations.
+
+```http
+HTTP/1.1 200 OK
+Content-Type: application/json
+
+[
+  {
+    "id":5646929551687680,
+    "updated":"2018-06-07T13:50:44Z",
+    "created":"2018-06-07T13:50:44Z",
+    "assignment":4873731453222912,
+    "doneBase":null,
+    "doneFull":null,
+    "recording":{
+       "recording":"5715999101812736",
+       "organisation":"aschool",
+       "challenge":"Assign_1_1_lijn3_thema9_06_EenzonvoorJet_a",
+       "digest":null,
+        "tenant":"demo"
+    },
+    "audioUrl":"https://api.itslanguage.nl/download/2257a2f8ada1446bae8d97f591169de9",
+    "fragments":[
+       {
+         "start":5.81,
+         "labels":[
+            {
+               "comment":"",
+               "layer":"Words",
+               "label":"speelt"
+            }
+         ],
+         "end":6.45
+      },
+      {
+         "start":6.67,
+         "labels":[
+            {
+               "comment":"",
+               "layer":"Words",
+               "label":"op"
+            }
+         ],
+         "end":7.17
+      }
+    ]
+  }
+]
+```
+
+
+## Get a single annotation
+
+### URL
+
+```http
+GET /annotation/:annotation HTTP/1.1
+```
+
+* `annotation` - **Required** The annotation identifier.
+
+### Request
+
+```http
+GET /annotation/annotation_1 HTTP/1.1
+Accept: application/json
+```
+
+### Response
+
+```http
+HTTP/1.1 200 OK
+Content-Type: application/json
+
+{
+  ...
+}
+```
+
+
+## Create an annotation
+
+### URL
+
+```http
+POST /annotations HTTP/1.1
+```
+
+### Request parameters
+
+Parameters should be posted as JSON body.
+
+Name             | Type     | Description
+-----------------|----------|------------
+`id`             | `string` | **Optional** The unique identier of the annotation. If none is given, one is generated.
+`doneBase`       | `string` | **Optional** The datetime this annotation was marked as done by the base annotator.
+`doneFull`       | `string` | **Optional** The datetime this annotation was marked as done by the full annotator.
+`recording`      | `string` | **Required** The reference to the recording (type
+`FlatRecordingReference` as specified below.
+`assignment`     | `string` | **Required** The unique identifier of the assignment this annotation
+is for.
+`fragments`      | `list`   | **Required** The annotated fragments (type `Fragment` as specified below).
+
+The Recording format (`FlatRecordingReference`).
+
+Parameters should be posted as JSON body.
+
+Name             | Type     | Description
+-----------------|----------|------------
+`tenant`         | `string` | **Required** The unique identier of the tenant.
+`organisation`   | `string` | **Required** The unique identifier of the organisation.
```

Comment:
For tenant and organisation:

- I am not sure tenant is really required
- It does not state which tenant and organisation. For the current user? admin user? or the user that made the recording?

Reply:
It is required, this code is used more often than we'd like:

```
    @classmethod                                                                
    def flat_reference_to_key(cls, reference):                                  
        """                                                                     
        Build a key from a :class:`domain.FlatRecordingReference`.              
                                                                                
        Args:                                                                   
            recording (.domain.FlatRecordingReference): The recording           
                reference to convert.                                           
                                                                                
        Returns:                                                                
            ndb.Key: The key for the given flat reference.                      
                                                                                
        """                                                                     
        org = try_to_coerce_into_int(reference.organisation)                    
        cha = try_to_coerce_into_int(reference.challenge)                       
        rec = try_to_coerce_into_int(reference.recording)                       
                                                                                
        organisation = ndb.Key(Organisation, org, namespace=reference.tenant)   
        challenge = ndb.Key(SpeechChallenge, cha, parent=organisation)          
        return ndb.Key(SpeechRecording, rec, parent=challenge)  
```

I think the SDK shields the user from these internals. I'd rather see a GUID kind of id for the audio recordings at some point. I'll chalk that up to its-gae refactoring tasks ;-)